File: solutions/3005. Count Elements With Maximum Frequency/3005.cpp
// Time complexity: O(n)
// Space complexity: O(n)
class Solution {
public:
    int maxFrequencyElements(const vector<int>& nums) {
        unordered_map<int, int> count;
        int maxCount = 0;
        int ans = 0;
        
        for (const int num : nums)
            ++count[num];
        
        for (const auto& pair : count)
            maxCount = max(maxCount, pair.second);
            
        for (const auto& pair : count)
            if (pair.second == maxCount) 
                ans += pair.second;    
            
        return ans;     
    }
};

File: solutions/3005. Count Elements With Maximum Frequency/3005.js
// Time complexity: O(n)
// Space complexity: O(n)
var maxFrequencyElements = function(nums) {
  const count = new Map();

  for (let num of nums)
    count.set(num, (count.get(num) || 0) + 1);

    const maxCount = Math.max(...count.values());
    let ans = 0;

    for (let value of count.values())
      if (value === maxCount)
        ans += value;

    return ans;    
};